src/java.base/share/classes/java/io/ObjectInputFilter.java

```diff
@@ -399,13 +765,97 @@ static ObjectInputFilter createFilter2(String pattern) {
+         * <p>
+         * Example, to create a filter that will allow any class loaded from the platform classloader.
+         * <pre><code>
+         *     ObjectInputFilter f = allowFilter(cl -> cl.getClassLoader() == ClassLoader.getPlatformClassLoader()
+         *                                          || cl.getClassLoader() == null, Status.UNDECIDED);
+         * </code></pre>
+         *
+         * @param predicate a predicate to test a non-null Class, non-null
+         * @param otherStatus a Status to use if the predicate is {@code false}
+         * @return {@link Status#ALLOWED} if the predicate on the class returns {@code true},
+         *          otherwise {@link Status#UNDECIDED}
+         */
+        public static ObjectInputFilter allowFilter(Predicate<Class<?>> predicate, Status otherStatus) {
+            Objects.requireNonNull(predicate, "predicate");
+            Objects.requireNonNull(otherStatus, "otherStatus");
+            return new Config.PredicateFilter(predicate, ALLOWED, otherStatus);
+        }
+
+        /**
+         * Returns a filter that returns {@code Status.REJECTED} if the predicate on the class is {@code true},
+         * otherwise the {@code otherStatus}.
+         *
+         * When the filter's {@link ObjectInputFilter#checkInput} method is invoked,
+         * the predicate is applied to the {@link FilterInfo#serialClass()}, the return Status is:
+         * <ul>
+         *     <li>{@link Status#UNDECIDED}, if the {@code serialClass} is {@code null},</li>
+         *     <li>{@link Status#REJECTED}, if the predicate on the class returns {@code true},</li>
+         *     <li>Otherwise, return {@code otherStatus}.</li>
+         * </ul>
+         * <p>
+         * Example, to create a filter that will reject any class loaded from the application classloader.
+         * <pre><code>
+         *     ObjectInputFilter f = rejectFilter(cl ->
+         *          cl.getClassLoader() == ClassLoader.ClassLoader.getSystemClassLoader(), Status.UNDECIDED);
+         * </code></pre>
+         *
+         * @param predicate a predicate to test a non-null Class, non-null
+         * @param otherStatus a Status to use if the predicate is {@code false}
+         * @return {@link Status#REJECTED} if the predicate on the class returns {@code true},
+         *          otherwise {@link Status#UNDECIDED}
+         */
+        public static ObjectInputFilter rejectFilter(Predicate<Class<?>> predicate, Status otherStatus) {
+            Objects.requireNonNull(predicate, "predicate");
+            Objects.requireNonNull(otherStatus, "otherStatus");
+            return new Config.PredicateFilter(predicate, REJECTED, otherStatus);
+        }
+
+        /**
+         * Returns a filter that returns {@code Status.ALLOWED} if the check is for limits
+         * and not checking a class; otherwise {@code Status.UNDECIDED}.
+         * If the {@link FilterInfo#serialClass()} is {@code null}, the filter returns
```

Comment:
"Returns a filter that returns {@code Status.ALLOWED} ... ". Maybe "Returns a filter that allows all limits ..".

Up-levelling a bit, it seems that filter operations fall into two broad categories: 1) class-checking, and 2)limits-checking. At least, in the way that some of these factories and adapters are laid out.  If we had such a notional at the class-level, then maybe that could influence the naming of these factories - and tie them to a higher-level concept.

Reply:
I'll add some higher level descriptions to the Config class and to ObjectInputFilter class doc.